metasv/generate_final_vcf.py

```diff
@@ -17,12 +17,175 @@
 vcf_template = os.path.join(mydir, "resources/template.vcf")
 
 
-def convert_metasv_bed_to_vcf(bedfile=None, vcf_out=None, vcf_template_file=vcf_template, sample=None, reference=None,
+def check_duplicates(interval1,interval2,max_dist=10):
+    if interval1.chrom != interval2.chrom or \
+       abs(interval1.start-interval2.start)>max_dist or \
+       abs(interval1.end-interval2.end)>max_dist or \
+       interval1.fields[3].split(",")[1] != interval2.fields[3].split(",")[1]:
+        return None
+    info1=json.loads(base64.b64decode(interval1.fields[3].split(",")[0]))
+    info2=json.loads(base64.b64decode(interval2.fields[3].split(",")[0]))
+    svmethods= sorted(list(set(info1["SVMETHOD"]+info2["SVMETHOD"])))
+    sources = []
+    if "SOURCES" in info1:
+        sources.append(info1["SOURCES"])
+    if "SOURCES" in info2:
+        sources.append(info2["SOURCES"])
+    sources=",".join(sources)
+    if sources:
+        info1["SOURCES"]=sources
+    if "PASS" in [interval1.fields[7], interval2.fields[7]] or ("AS" not in svmethods and len(set(svmethods) - {"SC","AS"})>1):
+        sv_filter = "PASS"
+    else:
+        sv_filter = "LowQual"
+            
+    end = max(interval1.end,interval2.end)
+    start = min(interval1.start,interval2.start)
+    info1.update(
+        {"END": end, "SVMETHOD": svmethods, "NUM_SVMETHODS": len(svmethods)})
+    return pybedtools.Interval(interval1.chrom, start,end, name="%s,%s,%d,%s" % (
+                               base64.b64encode(json.dumps(info1)), 
+                               info1["SVTYPE"], end-start,
+                               ";".join(svmethods)), 
+                               score = interval1.score, 
+                               otherfields=[interval1.fields[6],sv_filter])
+
+
+
+def get_interval_info(feature,pass_calls):
+    func_logger = logging.getLogger("%s" % (get_interval_info.__name__))
+    
+    pos = feature.start
+    end = feature.end
+    genotype = "./." if len(feature.fields) < 11 else feature.fields[10]
+    if genotype == "0/0":
+        func_logger.info("Skipping homozygous reference %s" % str(feature))
+        return None
+
+    sub_names = feature.name.split(":")
+    sub_lengths = map(lambda x: int(x.split(",")[2]), sub_names)
+
+    sub_types = map(lambda x: x.split(",")[1], sub_names)
+    sub_methods = [name.split(",")[3] for name in sub_names]
+    svmethods = (";".join([name.split(",")[3] for name in sub_names])).split(";")
+    try:
+        info = json.loads(base64.b64decode(name.split(",")[0]))
+    except TypeError:
+        info = dict()
+    if len(feature.fields) > 9:
+        info.update(json.loads(base64.b64decode(feature.fields[9])))
+
+    index_to_use = 0
+    is_pass = False
+    svlen = -1
+    if "DEL" in sub_types:
```

Comment:
I am not sure, if it happens in practice I think so far I assume we have separate lines for each SV type.